jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/servlet/FilterDiagnosticsCollector.java

```diff
@@ -1,134 +1,101 @@
+
+package org.eclipse.lsp4jakarta.jdt.core.servlet;
+
+import java.util.List;
+
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IMemberValuePair;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4jakarta.jdt.core.AbstractDiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.JakartaCorePlugin;
+
+public class FilterDiagnosticsCollector extends AbstractDiagnosticsCollector {
+
+    public FilterDiagnosticsCollector() {
+    	super();
+    }
+
+    @Override
+	protected String getDiagnosticSource() {
+		return ServletConstants.DIAGNOSTIC_SOURCE;
+	}
+
+    public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+        if (unit != null) {
+            IType[] alltypes;
+            IAnnotation[] allAnnotations;
+
+            try {
+                alltypes = unit.getAllTypes();
+                for (IType type : alltypes) {
+                    allAnnotations = type.getAnnotations();
+                    IAnnotation webFilterAnnotation = null;
+
+                    for (IAnnotation annotation : allAnnotations) {
+                        if (isMatchedJavaElement(type, annotation.getElementName(), ServletConstants.WEBFILTER_FQ_NAME)) {
+                            webFilterAnnotation = annotation;
+                        }
+                    }
+
+                    String[] interfaces = {ServletConstants.FILTER_FQ_NAME};
+                    boolean isFilterImplemented = doesImplementInterfaces(type, interfaces);
+                    
+                    if (webFilterAnnotation != null && !isFilterImplemented) {
+                        diagnostics.add(createDiagnostic(type, unit, 
+                        		"Annotated classes with @WebFilter must implement the Filter interface.", 
+                        		ServletConstants.DIAGNOSTIC_CODE_FILTER, null, ServletConstants.SEVERITY));
+                    }
+
+                    /* URL pattern diagnostic check */
+                    if (webFilterAnnotation != null) {
+                        IMemberValuePair[] memberValues = webFilterAnnotation.getMemberValuePairs();
+
+                        boolean isUrlpatternSpecified = false;
+//                        boolean isServletNamesSpecified = false;
+                        boolean isValueSpecified = false;
+                        for (IMemberValuePair mv : memberValues) {
+                            if (mv.getMemberName().equals(ServletConstants.URL_PATTERNS)) {
+                                isUrlpatternSpecified = true;
+                                continue;
+                            }
+//                            if (mv.getMemberName().equals(ServletConstants.SERVLET_NAMES)) {
+//                                isServletNamesSpecified = true;
+//                                continue;
+//                            }

```

Comment:
Why is this commented out?

Reply:
The following error message doesn't mention the field, not sure what does spec. say.

Reply:
I'm not sure I follow, which following error message?

Reply:
line 87, `The annotation @WebServlet must define the attribute urlPatterns or the attribute value.`

Reply:
Found this from https://tomcat.apache.org/tomcat-10.0-doc/servletapi/jakarta/servlet/annotation/WebFilter.html, don't see it mentions `servletNames`

`At least one URL pattern MUST be declared in either the value or urlPattern attribute of the annotation, but not both.

The value attribute is recommended for use when the URL pattern is the only attribute being set, otherwise the urlPattern attribute should be used.`

Reply:
Looked into this a bit more, looks like you caught a mistake. The diagnostic message should be for `@WebFilter` and it was accidentally changed to `@WebServlet` in this PR: https://github.com/eclipse/lsp4jakarta/commit/da86428e6ee935fe6cdb186c16eb8b096eb7eb9d#diff-a2ea83e5123f5ab5d237167e4336120ac1ccfa4ca03f4a02394631cde36c7619R113

We should keep the servlet names check as from the Servlet spec: https://jakarta.ee/specifications/servlet/5.0/jakarta-servlet-spec-5.0.html#_WebFilter:
"The urlPatterns attribute, servletNames attribute or the value attribute of the annotation MUST be specified"

Reply:
Added it back in